libs/cardano-ledger-core/testlib/Test/Cardano/Ledger/Core/JSON.hs

```diff
@@ -68,3 +70,14 @@ goldenJsonPParamsSpec =
   it "Golden JSON specs for PParams " $ \file -> do
     decoded <- eitherDecodeFileStrict @(PParams era) file
     void $ expectRightExpr decoded
+
+goldenJsonPParamsUpdateSpec ::
+  forall era.
+  EraTest era =>
+  SpecWith FilePath
+goldenJsonPParamsUpdateSpec =
+  it "Golden JSON specs for PParamsUpdate" $ \file -> do
+    let ppu = runGen 100 100 (arbitrary @(PParamsUpdate era))
+    let encoded = T.unpack $ T.decodeUtf8 $ BSL.toStrict $ encodePretty ppu
+    fileContent <- T.unpack . T.decodeUtf8 . BSL.toStrict <$> BSL.readFile file
+    encoded `shouldBe` fileContent
```

Comment:
It is ok to compare `Text`, since JSON is a textual format and in case there are any unicode characters they will be showed in the diff correctly if there are any failures
```suggestion
    let encoded = T.decodeUtf8 $ BSL.toStrict $ encodePretty ppu
    fileContent <- T.decodeUtf8 . BSL.toStrict <$> BSL.readFile file
    encoded `shouldBe` fileContent
```